inat.label.py

```
@@ -292,51 +344,147 @@ def extract_observation_id(input_string, debug = False):
-            if retries > 0:
-                print("Rate limit exceeded. Waiting 5 seconds before retry...")
-                time.sleep(5)
-                return fetch_api_data(url, retries - 1)
-            else:
-                return None, "Max retries reached"
-        else:
-            return None, f"Status code: {response.status_code}"
+                from email.utils import parsedate_to_datetime
+                dt = parsedate_to_datetime(ra)
+                return max(0.0, (dt - datetime.datetime.utcnow()).total_seconds())
+            except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)